projects/plugins/jetpack/extensions/plugins/post-publish-qr-post-panel/index.js

```diff
@@ -16,10 +20,34 @@ const QRIcon = () => (
 	</SVG>
 );
 
+/**
+ * React component that renders a QR code for the post,
+ * pulling the post data from the editor store.
+ *
+ * @param {object} props      - Component props.
+ * @returns {React.Component}   The react component.
+ */
+function QRPost( props ) {
+	const {
+		post: { title },
+		permalink,
+	} = useSelect(
+		select => ( {
+			post: select( editorStore ).getCurrentPost(),
+			permalink: select( editorStore ).getPermalink(),
+		} ),
+		[]
+	);
+
+	const codeContent = `${ title } ${ permalink }`;
+
+	return <QRCode value={ codeContent } size={ 248 } renderAs="canvas" { ...props } />;
```

Comment:
just in case, we're aware that `{ ...props }` can be considered an antipattern but it's a private component for now, not externally so I think we're ok.